### PR TITLE
chore(Bonus Pagamenti Digitali): [IAC-130] Add bottom sheet to display the payment method to delete

### PR DESCRIPTION
**Pull request by Undermaken**
## Short description
This PR adds the bottomsheet that shows those payment method that will be deleted in the opt-in flow.

https://user-images.githubusercontent.com/822471/153284828-25af15af-ea09-417d-b873-5e53c75afc60.mov

## List of changes proposed in this pull request
- Add the bottomsheet component and hook
- Add locales
- Add tests

## How to test
- add some payment method compatible with BPD in you wallet
- open the bottomsheet in this way
```typescript
const MyAwsomeComponent = () => {
  const { presentBottomSheet } = useBottomSheetMethodsToDelete({
    onDeletePress: () => console.log("onDeletePress"),
    onCancelPress: () => console.log("onCancelPress")
  });
  return (
    <ButtonDefaultOpacity onPress={presentBottomSheet}>
      <Label>{"Open Bottom Sheet"}</Label>
    </ButtonDefaultOpacity>
  );
};
```

### CODE REVIEW

**Comment by thisisjp** [comment]
@Undermaken let me dream: can the bottom sheet have a dynamic height depending on the number of displayed methods—up to a maximum height just right below the main header? By doing this, we don't have to scroll all the time, only when really needed :)

**Comment by Undermaken** [comment]
> @Undermaken let me dream: can the bottom sheet have a dynamic height depending on the number of displayed methods—up to a maximum height just right below the main header? By doing this, we don't have to scroll all the time, only when really needed :) 

I already implemented this logic: the bottomsheet grows with the number of items but it never exceeds the 70% (we can customize it) of the height of screen

Sounds good?

**Comment by Undermaken** [comment]
new update

https://user-images.githubusercontent.com/822471/153460451-f336526f-8b7b-4895-b9e5-2651c4ce9d16.mov

File: ts/features/bonus/bpd/components/optInStatus/BottomSheetMethodsToDelete.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            {I18n.t(
+              "bonus.bpd.optInPaymentMethods.deletePaymentMethodsBottomSheet.title"
+            )}
+          </H3>
+          <View spacer={true} />
+        </View>
+      );
+    }
+  };
+};
```
Comment (CrisTofani) [inline ts/features/bonus/bpd/components/optInStatus/BottomSheetMethodsToDelete.tsx:111]:
```suggestion
export const useBottomSheetMethodsToDelete = (
  props: Pick<Props, "onDeletePress" | "onCancelPress">
): BottomSheetReturnType => {
  const paymentMethods = useSelector(getBPDMethodsSelector);
  const snapPoint = Math.min(
    Dimensions.get("window").height * 0.8,
    // (subtitle + footer) + items
    280 + paymentMethods.length * 58
  );
  
  const bottomSheetContent = (
    <BottomSheetMethodsToDelete
      paymentMethods={paymentMethods}
      onDeletePress={props.onDeletePress}
      onCancelPress={props.onCancelPress}
    />
  );

  const bottomSheetHeader = (
    <View style={IOStyles.flex}>
      <H3>
        {I18n.t(
          "bonus.bpd.optInPaymentMethods.deletePaymentMethodsBottomSheet.title"
        )}
      </H3>
      <View spacer={true} />
    </View>
  );

  const { present: presentBottomSheet } = useIOBottomSheet(
    bottomSheetContent,
    bottomSheetHeader,
    snapPoint
  );

  return {
    presentBottomSheet
  };
};
```
Just a suggestion, mainly helpful for the bottom-sheet library update. In this case the usage of `useIOBottomSheet` is enough what do you think? (didn't tested it)

**Review by CrisTofani — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/components/optInStatus/BottomSheetMethodsToDelete.tsx (hunk 1 above)
Comment (Undermaken) [reply ts/features/bonus/bpd/components/optInStatus/BottomSheetMethodsToDelete.tsx:111]: I guess it doesn't work.
I'm pretty sure I tried your approach and I found some down sides
This is what happen

### your suggestion
https://user-images.githubusercontent.com/822471/153578742-fa88a832-2fd5-419c-aa8c-61e5779d76e9.mov

### current
https://user-images.githubusercontent.com/822471/153578804-5eaaddfe-8f12-454b-b60b-768a7c7c1b90.mov

Comment (CrisTofani) [reply ts/features/bonus/bpd/components/optInStatus/BottomSheetMethodsToDelete.tsx:111]: not at the moment, but the footer is integrated in [`react-native-bottom-sheet` v4](https://gorhom.github.io/react-native-bottom-sheet/components/bottomsheetfooter)

We will solve even this issue 😄

**Pull request merged**